File: 744-find-smallest-letter-greater-than-target/744-find-smallest-letter-greater-than-target.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int st = 0, en = letters.size()-1, md, ans = 0;
        while(en>=st){
            md = (st+en)/2;
            if(letters[md]>target) ans = md, en = md-1;
            else st = md+1;
        }
        return letters[ans];
    }
};

File: 744-find-smallest-letter-greater-than-target/744-find-smallest-letter-greater-than-target.cs
public class Solution {
    
    public char NextGreatestLetter(char[] letters, char target) {
        int st = 0, en = letters.Length -1, md = 0;
        char ans = letters[0];
        while(en>=st){
            md = (en+st)/2;
            if(letters[md] > target){
                en = md-1;
                ans = letters[md];
            } 
            else st = md+1;
        }
        return ans;
    }
}
